Sorts/TimSort.java

```diff
@@ -0,0 +1,177 @@
+package Sorts;
+
+import java.lang.Math;
+import java.util.Random;
+
+/**
+ * @author Hemanth Kotagiri (https://github.com/hemanth-kotagiri)
+ * @see SortAlgorithm
+ */
+
+ // Wikipedia: https://en.wikipedia.org/wiki/Tim_sort
+class TimSort {
+    int array[];
+    int array_length;
+    int RUN = 32;
+
+    /**
+     * A constructor which takes in the array specified by the user.
```

Comment:
use `@brief` for brief description of function or class whatever
```suggestion
     * @brief A constructor which takes in the array specified by the user.
```